cra-rxjs-styled-components/src/components/repo-meta/RepoMeta.tsx

```diff
@@ -0,0 +1,53 @@
+import { formatDistance } from 'date-fns';
+import { getColourForLanguage } from '../../helpers/colours';
+import { BranchLogo } from '../misc/BranchLogo';
+import { StarLogo } from '../misc/StarLogo';
+import MetaIcon from './MetaIcon';
+import { LanguageColor, Metadata, SocialWrapper } from './RepoMeta.styles';
+
+interface RepoMetaProps {
+  language?: string | null;
+  stargazerCount: number;
+  forkCount: number;
+  updatedAt: Date;
+}
+
+function RepoMeta({
+  language,
+  stargazerCount,
+  forkCount,
+  updatedAt,
+}: RepoMetaProps) {
+  return (
+    <Metadata>
+      {language && (
+        <div>
+          <LanguageColor
+            style={{
+              backgroundColor: getColourForLanguage(language) || '#ccc',
+            }}
+          />
+          {language}
+        </div>
+      )}
+      {(stargazerCount > 0 || forkCount > 0) && (
+        <SocialWrapper>
+          {stargazerCount > 0 && (
+            <MetaIcon text={stargazerCount} children={<StarLogo />} />
+          )}
+          {forkCount > 0 && (
+            <MetaIcon text={forkCount} children={<BranchLogo />} />
+          )}
```

Comment:
So I still have an issue with this specifically... I don't like having logic wrapping logic for rendering. I'd almost prefer before the return to build the social icons and then inject them here instead or defer some of the render logic down a layer. So we could do something like:

```
const socialIcons = [];
if (stargazerCount > 0) {
  socialIcons.push(<MetaIcon text={stargazerCount} children={<StarLogo />} />);
}
if (forkCount > 0) {
  socialIcons.push(<MetaIcon text={forkCount} children={<StarLogo />} />);
}

return (
   // ...
  {socialIcons.length > 0 && (<SocialWrapper>{socialIcons}</SocialWrapper>)}
);
```

or something that moves the logic out of the view like this.

Reply:
I think this is neat

Reply:
Glad you liked it! It's mostly a readability thing. I have the same issue with nested ternaries. If I see layered logic, I see a smell for someone later on. What's cool about this too is we could theoretically add another section if it exists.